src/buildblock/FilePath.cxx

```diff
@@ -0,0 +1,386 @@
+#include "stir/FilePath.h"
+
+START_NAMESPACE_STIR
+
+FilePath::FilePath(){
+    initSeparator();
+}
+
+FilePath::FilePath(const std::string &__str, bool _check)
+{
+    if(__str.size() == 0)
+        error(boost::format("FilePath: Cannot initialise empty path."));
+
+    my_string = __str;
+
+    initSeparator();
+
+    if (!_check)
+        return;
+
+    // Checks
+    checks();
+    // Check permissions
+}
+
+bool FilePath::is_directory() const
+{
+    struct stat info;
+
+    if( stat( my_string.c_str(), &info ) != 0 )
+        error(boost::format("FilePath: Cannot access %1%.")%my_string);
+    else if( info.st_mode & S_IFDIR )
+        return true;
+
+    return false;
+}
+
+
+bool FilePath::is_regfile() const
+{
+    struct stat info;
+
+    if( stat( my_string.c_str(), &info ) != 0 )
+        error(boost::format("FilePath: Cannot access %1%.")%my_string);
+    else if( info.st_mode & S_IFREG )
+        return true;
+    else
+        return false;
+}
+
+bool FilePath::is_writable() const
+{
+#if defined(__OS_WIN__)
+	return true; 
+#else	
+    if( access(my_string.c_str(), 0) == 0 )
+        return true;
+    else
+        return false;
+#endif
+}
+
+bool FilePath::exist(std::string s)
+{
+    struct stat info;
+
+    if (s.size()>0)
+    {
+        if( stat( s.c_str(), &info ) != 0 )
+            return false;
+        else
+            return true;
+    }
+
+    //    if( stat( my_string.c_str(), &info ) != 0 )
+    //        return false;
+    //    else
+    //        return true;
+
+}
+
+FilePath FilePath::append(FilePath p)
```

Comment:
argument should be `const FilePath&`. Does the function change the current member? If so, it shouldn't return a `FilePath`. If not, it has to be a `const` member.